external/src/main/java/com/redhat/cloud/policies/engine/actions/plugins/NotificationActionPluginListener.java

```diff
@@ -54,14 +56,35 @@
     @Metric(absolute = true, name = "engine.actions.webhook.processed")
     Counter messagesCount;
 
+    class PoliciesParamsBuilder {
+        private Map<String, String> triggers;
+
+        public PoliciesParamsBuilder() {
+            this.triggers = new HashMap<>();
+        }
+
+        public PoliciesParamsBuilder addTrigger(String key, String value) {
+            this.triggers.put(key, value);
+            return this;
+        }
+
+        public String build() throws JsonProcessingException {
+            Map<String, Object> params = new HashMap<>();
+            params.put("triggers", this.triggers);
+            ObjectMapper mapper = new ObjectMapper();
```

Comment:
Depending on how often this is called, it could be pulled out and initialised in the constructor